charts/simple-app/templates/prometheusrules.yaml

```diff
@@ -0,0 +1,265 @@
+          container {{`{{`}} $labels.container{{`}}`}} has been in waiting
+          state for longer than 1 hour.
+      expr: sum by (namespace, pod, container) (kube_pod_container_status_waiting_reason{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}) > 0
+      for: {{ .for }}
+      labels:
+        severity: {{ .severity }}
+        {{- with $values.prometheusRules.additionalRuleLabels }}
+        {{ toYaml . | nindent 8 }}
+        {{- end }}
+    {{- end }}
+
+  - name: {{ .Release.Namespace }}.{{ .Release.Name }}.{{ .Chart.Name }}.DeploymentRules
+    rules:
+
+    {{ with .Values.prometheusRules.DeploymentGenerationMismatch -}}
+    - alert: DeploymentGenerationMismatch
+      annotations:
+        summary: Deployment generation mismatch due to possible roll-back
+        runbook_url: {{ $runbookUrl }}#deploymentgenerationmismatch
+        description: >-
+          Deployment generation for {{`{{`}} $labels.namespace
+          {{`}}`}}/{{`{{`}} $labels.deployment {{`}}`}} does not match, this
+          indicates that the Deployment has failed but has not been rolled
+          back.
+      expr: |-
+        kube_deployment_status_observed_generation{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}
+          !=
+        kube_deployment_metadata_generation{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}
+      for: {{ .for }}
+      labels:
+        severity: {{ .severity }}
+        {{- with $values.prometheusRules.additionalRuleLabels }}
+        {{ toYaml . | nindent 8 }}
+        {{- end }}
+    {{- end }}
+
+    {{ with .Values.prometheusRules.DeploymentReplicasMismatch -}}
+    - alert: DeploymentReplicasMismatch
+      annotations:
+        summary: Deployment has not matched the expected number of replicas.
+        runbook_url: {{ $runbookUrl }}#deploymentreplicasmismatch
+        description: >-
+          Deployment {{`{{`}} $labels.namespace {{`}}`}}/{{`{{`}}
+          $labels.deployment {{`}}`}} has not matched the expected number of
+          replicas for longer than {{ .for }}.
+      expr: |-
+        (
+          kube_deployment_spec_replicas{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}
+            !=
+          kube_deployment_status_replicas_available{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}
+        ) and (
+          changes(kube_deployment_status_replicas_updated{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}[5m])
+            ==
+          0
+        )
+      for: {{ .for }}
+      labels:
+        severity: {{ .severity }}
+        {{- with $values.prometheusRules.additionalRuleLabels }}
+        {{ toYaml . | nindent 8 }}
+        {{- end }}
+    {{- end }}
+
+  {{- if .Values.autoscaling.enabled }}
+  - name: {{ .Release.Namespace }}.{{ .Release.Name }}.{{ .Chart.Name }}.HorizontalPodAutoscalerRules
+    rules:
+
+    {{ with .Values.prometheusRules.HpaReplicasMismatch -}}
```

Comment:
maybe should have `and .Values.autoscaling.enabled`?

Reply:
@sabw8217 Thinking about this more ... I think this should be a global alarm. I can't think of a situation where a developer would have an autoscaling HPA in place, and want _different_ alarms than this?